### PR TITLE
Migrate the payee map to the rewards module

### LINKED ISSUE #359: Reintroduce block reward incentives
Since there's no slashing if a validator misses its slot of authoring a block, the rewards to validators should remain proportional to the number of the blocks they have authored in an era. We need to take the staking reward points on events like note_author, note_uncle coming from the pallet_authorship into account when calculating rewards for each validator.

### PR DESCRIPTION
**Pull request by aliXsed**
Closes #359

### CODE REVIEW

File: crml/staking/src/lib.rs [hunk 1]
```diff
@@ -771,6 +789,46 @@ decl_storage! {
 	}
 }
 
+fn migrate_payees_to_rewards_module<T: Trait>() -> frame_support::weights::Weight {
+	#[allow(dead_code)]
+	mod inner {
+		use codec::{Decode, Encode};
+		use sp_std::vec::Vec;
+		#[derive(Encode, Decode, Default)]
```
Comment (jordy25519) [inline crml/staking/src/lib.rs:799]: nit: is it possible to import the real type definition here?

**Review by jordy25519 — COMMENTED** [review]
generally lgtm, I guess it's possible to add a unit test for the behaviour e.g. https://github.com/paritytech/substrate/pull/6334/files#diff-ac5a01a5421b9a014029356743573450768a049c4c727857e0d4db55e6b3f003R744-R773

File: crml/staking/src/lib.rs (hunk 1 above)
Comment (aliXsed) [reply crml/staking/src/lib.rs:799]: Can you please explain a bit more?

Comment (jordy25519) [reply crml/staking/src/lib.rs:799]: `OldEraPoints`?
should it be imported `types::EraRewardPoints`

**Comment by aliXsed** [comment]
> generally lgtm, I guess it's possible to add a unit test for the behaviour e.g. https://github.com/paritytech/substrate/pull/6334/files#diff-ac5a01a5421b9a014029356743573450768a049c4c727857e0d4db55e6b3f003R744-R773

A test has been added for this purpose.

File: crml/staking/src/lib.rs (hunk 1 above)
Comment (aliXsed) [reply crml/staking/src/lib.rs:799]: I have removed the old structure definition. The new definition is internally different from the old one and follows the latest as in substrate.

File: crml/staking/src/lib.rs [hunk 2]
```diff
@@ -621,6 +621,21 @@ impl Default for Forcing {
 	}
 }
 
+// A value placed in storage that represents the current version of the Staking storage. This value
+// is used by the `on_runtime_upgrade` logic to determine whether we run storage migration logic.
+// This should match directly with the semantic versions of the Rust crate.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug)]
+enum Releases {
+	V0 = 0,
+	V1 = 1,
```
Comment (jordy25519) [inline crml/staking/src/lib.rs]: could you add a comment about the runtime versions for V0 and V1

**Review by jordy25519 — APPROVED** [review]

File: crml/staking/src/lib.rs (hunk 2 above)
Comment (aliXsed) [reply crml/staking/src/lib.rs]: Sure

**Pull request merged**